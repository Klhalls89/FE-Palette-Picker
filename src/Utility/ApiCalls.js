import {baseUrl} from './Config/BaseUrl';

export const fetchProjects = async () => {
  const url = 'http://localhost:3001/api/v1/projects/';
  const response = await fetch(url);

  if (response.status == 200) {
    const body = await response.json();

    return body;
  } 
};

export const newProject = async (projectName) => {
  const url = `${baseUrl}/api/v1/projects`;
  const options = {
    method: 'POST',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(projectName)
  };

  const response = await fetch(url, options);
  if (!response.ok) {
    return response.statusText;
  }
};