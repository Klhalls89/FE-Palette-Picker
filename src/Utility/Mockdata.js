const mockProjects = [{
  id: 1,
  project_title: 'cheers'
}]

export {
  mockProjects
}