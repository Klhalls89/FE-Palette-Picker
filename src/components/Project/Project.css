.project-area {
  text-align: center;
}

.proj-name {
  text-align: center;
}

.saved-palette-disp {
  display: flex;
  align-items: center;
  justify-content: center;
  border: solid #fff 3px;
}

.saved-palette-disp h3 {
  font-family: 'Courgette', cursive;
}

.saved-color1:hover,
.saved-color2:hover,
.saved-color3:hover,
.saved-color4:hover,
.saved-colors5:hover {
  transform: scale(1.2);
}

.far {
  margin: 5px;
  font-size: 38px;
  padding: 0px 10px;
  margin: auto;
}

.far:hover {
  color: grey;
  transform: scale(1.2);
}

.fas {
  margin: 5px;
  font-size: 36px;
  padding-left: 17px;
  margin: auto;
}

.fas:hover {
  color: grey;
  transform: scale(1.2);
}

.saved-colors {
  grid-area: colors;
  display: flex;
}

.saved-colors div {
  background-color: pink;
  height: 50px;
  width: 50px;
  margin: 5px;
}

.saved-palette-disp:hover {
  border: solid #3C3C3C 3px;
  cursor: pointer;
}
