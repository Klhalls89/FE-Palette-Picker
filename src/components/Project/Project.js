import React from 'react';

const Project = ({ 
  palette_title, project_title, 
  id, color_1, color_2, color_3, color_4, color_5
}) => {

  if (id) { 
    return (
      <article className="project-area">
        <h2 className="proj-name">
          {project_title}
        </h2>
        <div className="saved-palette-disp">
          <h3 className="pal-name">
            {palette_title}
          </h3>
          <i className="fas fa-trash-alt"></i>
          <section className="saved-colors">
            <div style={{backgroundColor: `#${color_1}`}} 
              className="saved-color1">
            </div>
            <div style={{backgroundColor: `#${color_2}`}} 
              className="saved-color2">
            </div>
            <div style={{backgroundColor: `#${color_3}`}} 
              className="saved-color3">
            </div>
            <div style={{backgroundColor: `#${color_4}`}} 
              className="saved-color4">
            </div>
            <div style={{backgroundColor: `#${color_5}`}} 
              className="saved-color5">
            </div>
          </section>
        </div>
      </article>
    );
  } else {
    return (
      <div>
        <h1></h1>
      </div>
    );
  }


  
};
export default Project;