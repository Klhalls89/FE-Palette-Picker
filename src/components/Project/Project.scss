  .project-area {

    .proj-name {
      text-align: center;
    } 
    
    .saved-palette-disp {
      display: flex;
      align-items: center;
      justify-content: center;
      border: solid #fff 3px;

      h3 {
        font-family: $accent-font;
      }
      .saved-color1:hover {
        transform: scale(1.2)
      }
      .saved-color2:hover {
        transform: scale(1.2)
      }
      .saved-color3:hover {
        transform: scale(1.2)
      }
      .saved-color4:hover {
        transform: scale(1.2)
      }
      .saved-color5:hover {
        transform: scale(1.2)
      }
      .far {
        margin: 5px;
        font-size: 38px;
        padding: 0px 10px;
        margin: auto;
        &:hover {
          color: grey;
          transform: scale(1.2)
        }
      }
      .fas {
        margin: 5px;
        font-size: 36px;
        padding-left: 17px;
        margin: auto;
        &:hover {
          color: grey;
          transform: scale(1.2)
        }
      }
    }

    .saved-colors {
      grid-area: colors;
      display: flex;

      div {
        background-color: pink;
        height: 50px;
        width: 50px;
        margin: 5px;
      }
    }

    .saved-palette-disp:hover {
      border: solid #3C3C3C 3px;
      cursor: pointer;
    }
  }
