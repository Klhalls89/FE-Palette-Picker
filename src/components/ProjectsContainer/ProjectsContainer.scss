.projects {
  border-top: 4px #C3C3C3 solid;
}

.saved-portion {
  display: flex;
  flex-direction: column;
  display: grid;
  grid-template-columns: repeat(2 1fr);
}
