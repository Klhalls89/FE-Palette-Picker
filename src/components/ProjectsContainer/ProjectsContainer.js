import React, { Component } from 'react';
import Project from '../Project/Project'
import { fetchInfo } from '../../Utility/ApiCalls';

class ProjectsContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      palettes: []
    };
  };

  componentDidMount(){
    this.getPalettes()
  }

  getPalettes = async () => {
    const palettes = await fetchInfo('http://localhost:3001/api/v1/palettes/')
    this.setState({palettes})
  }

  displayProjects = () => {
    const { projects } = this.props
    const { palettes } = this.state

    const newProjects = projects.map(project => {
      const matchingPalettes = palettes.filter(palette => {
        return palette.project_id === project.id
      })
      return <Project project={project} palettes={matchingPalettes}  />
    })
    return newProjects
  }

  render() {
    const projectsToDisplay = this.displayProjects()
    return (
      <div className="projects">
        <form className="project-form">
          <input type="text"/>
          <button>Project Name</button>
        </form>
        <section className="saved-portion">
        {projectsToDisplay}
        </section>
      </div>
    )
  }
}

export default ProjectsContainer;