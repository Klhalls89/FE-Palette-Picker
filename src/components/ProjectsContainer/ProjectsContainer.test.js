import React from 'react';
import ProjectsContainer from './ProjectsContainer.js';
import Project from '../Project/Project.js'; 
import { removePalette } from '../../Utility/ApiCalls';
import { shallow } from 'enzyme';

let shortid = require('shortid');

describe('Projects Containers', () => {
  let wrapper;
  let mockEditPalette;
  let mockProjects;
  let mockAllPalettes;
  let mockRemovePalette;

  beforeEach(() => {
    mockEditPalette = jest.fn()
    mockRemovePalette = jest.fn()
    mockProjects = [{title: 'project one'}, {title: 'project two'}]
    mockAllPalettes = [{color:'123123'},{color:'123123'}]

    wrapper = shallow(<ProjectsContainer editPalette={mockEditPalette} 
                                          projects={mockProjects} 
                                          allPalettes={mockAllPalettes}  
                                          removePalette={mockRemovePalette}/>)
  });

  it('Projects component should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('deletePalette', () => {
    it.skip('deletePalette should call removePalette', () => {
      let mockFn = jest.spyOn(wrapper.instance(), 'removePalette');
      wrapper.instance().deletePalette(); 
      expect(mockRemovePalette).toHaveBeenCalled();
    });
  });
});

 
  