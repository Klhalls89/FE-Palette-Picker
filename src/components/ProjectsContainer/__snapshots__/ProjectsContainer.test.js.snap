// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Containers Projects component should match the snapshot 1`] = `
<div
  className="projects"
>
  <section
    className="saved-portion"
  >
    <Project
      color="123123"
      deletePalette={[Function]}
      editPalette={[MockFunction]}
      key="7nSOWcDTz"
      title="project one"
    />
    <Project
      color="123123"
      deletePalette={[Function]}
      editPalette={[MockFunction]}
      key="ubrYjlBp3z"
      title="project one"
    />
    <Project
      color="123123"
      deletePalette={[Function]}
      editPalette={[MockFunction]}
      key="7t55OyZni7"
      title="project two"
    />
    <Project
      color="123123"
      deletePalette={[Function]}
      editPalette={[MockFunction]}
      key="-NSNnlTXXr"
      title="project two"
    />
  </section>
</div>
`;
